src/parser/mod.rs

```diff
@@ -5153,6 +5167,19 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse configuration like primary key information during clickhouse table creation.
+    /// <https://clickhouse.com/docs/en/sql-reference/statements/create/table#primary-key>
+    fn parse_optional_click_house_create_table_config(
+        &mut self,
+    ) -> Result<ClickHouseTableConfiguration, ParserError> {
+        let mut primary_key = None;
+        if self.parse_keywords(&[Keyword::PRIMARY, Keyword::KEY]) {
+            primary_key = Some(Box::new(self.parse_expr()?));
+        };
+
+        Ok(ClickHouseTableConfiguration { primary_key })
+    }
```

Comment:
would it make sense to inline this similar to how engine is handled? (thinking since currently the function is only looking for primary key which would probably not be unique to clickhouse)

Reply:
Rid of this function and executes parsing of `PRIMARY KEY` in the function `parse_create_table` as other keywords.